scripts/prepare_int_data.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# we're using python 3.x style print but want it to work in python 2.x,
+from __future__ import print_function
+import re, os, argparse, sys, math, warnings, subprocess, shutil, threading
+
+parser = argparse.ArgumentParser(description="This program uses the vocabulary" 
+                                 "file to turn the data into ASCII-integer form," 
+                                 "and to give it a standard format in preparation"
+                                 " for language model training.")
+
+parser.add_argument("--fold-dev-into", type=str,
+                    help="If supplied, the name of data-source into which to fold the "
+                    "counts of the dev data for purposes of vocabulary estimation "
+                    "(typically the same data source from which the dev data was "
+                    "originally excerpted).")
+parser.add_argument("--parallel", type=str, choices=["false", "true"], default="true",
+                    help="Setting --parallel false will disable the (default) "
+                    "parallel processing of multiple data sources by this script.")
+parser.add_argument("text_dir", 
+                    help="Directory of original data")
+parser.add_argument("vocab", 
+                    help="Vocabulary")
+parser.add_argument("int_dir", 
+                    help="Directory of formed data")
+
+args = parser.parse_args()
+
+# Add the script dir and the src dir to the path.
+os.environ['PATH'] = (os.environ['PATH'] + os.pathsep +
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + os.pathsep +
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + "/../src");
+
+# do some checking
+if os.system("validate_text_dir.py " + args.text_dir) != 0:
+  sys.exit("prepare_int_data.py: failed to validate text directory")
+
+if os.system("validate_vocab.py " + args.vocab) != 0:
+  sys.exit("prepare_int_data.py: failed to validate vocab")
+
+if not os.path.exists(os.path.abspath(os.path.dirname(sys.argv[0])) + "/text_to_int.py"):
+  sys.exit("prepare_int_data.py: expected text_to_int.py to be on the path")
+
+# create the output data directory
+if not os.path.exists(args.int_dir):
```

Comment:
you can remove this if-block... it's redundant with the next line. [makedirs makes it recursively.]